# Project: Online Retail Store

## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#wrangling">Data Wrangling</a></li>
<li><a href="#Questions for Analysis">Questions for Analysis</a></li>
<li><a href="#Data Visualization">Data Visualization</a></li>
<li><a href="#conclusions">Conclusion</a></li>
</ul>

<a id='intro'></a>
## Introduction

### Project Description 

An online retail store has needed to review their data and provide insights that would be valuable to the CEO and CMO of the business. The business has been performing well and the management wants to analyze what the major contributing factors are in the revenue so they can strategically plan for next year.

The leadership is interested in viewing the metrics from both an operations and marketing perspective. Management also intends to expand the business and is interested in seeking guidance into areas that are performing well so they can keep a clear focus on what’s working. They would also like to view different metrics based on the demographic information that is available in the data.

Note: Stakeholders don't need "UK" in their analysis




<a id='wrangling'></a>
## Data Wrangling



### General Properties

Let's explore the dataset to know its properties

In [42]:
#Importing the needed packages
import pandas as pd 

In [43]:
#Reading our database
df= pd.read_excel('Online Retail.xlsx')

In [44]:
df.head()

InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

   UnitPrice  Total         InvoiceDate  CustomerID         Country  
0       2.55  15.30 2010-12-01 08:26:00     17850.0  United Kingdom  
1       3.39  20.34 2010-12-01 08:26:00     17850.0  United Kingdom  
2       2.75  22.00 2010-12-01 08:26:00     17850.0  United Kingdom  
3       3.39  20.34 2010-12-01 08:26:00     17850.0  United Kingdom  
4       3.39  20.34 2010-12-01 08:26:00     17850.0  United Kingdom

In [45]:
#Showing the shape of the dastaset
df.shape

(541909, 9)

In [46]:
#Showing some descriptive statistics about the dataset
df.describe()

Quantity      UnitPrice          Total     CustomerID
count  541909.000000  541909.000000  541909.000000  406829.000000
mean        9.552250       4.611114      17.987795   15287.690570
std       218.081158      96.759853     378.810824    1713.600303
min    -80995.000000  -11062.060000 -168469.600000   12346.000000
25%         1.000000       1.250000       3.400000   13953.000000
50%         3.000000       2.080000       9.750000   15152.000000
75%        10.000000       4.130000      17.400000   16791.000000
max     80995.000000   38970.000000  168469.600000   18287.000000

In [47]:
#Showing info about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 9 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   InvoiceNo    541909 non-null  object        
 1   StockCode    541909 non-null  object        
 2   Description  540455 non-null  object        
 3   Quantity     541909 non-null  int64         
 4   UnitPrice    541909 non-null  float64       
 5   Total        541909 non-null  float64       
 6   InvoiceDate  541909 non-null  datetime64[ns]
 7   CustomerID   406829 non-null  float64       
 8   Country      541909 non-null  object        
dtypes: datetime64[ns](1), float64(3), int64(1), object(4)
memory usage: 37.2+ MB


In [48]:
#Count NULL values in the columns
df.isnull().sum()

InvoiceNo           0
StockCode           0
Description      1454
Quantity            0
UnitPrice           0
Total               0
InvoiceDate         0
CustomerID     135080
Country             0
dtype: int64

In [49]:
#Count the duplicated rows
df.duplicated().sum()

5268

### Data Cleaning
It's the time to clean our dataset 

In [50]:
#Dropping the duplicates
df.drop_duplicates(inplace=True)
df.duplicated().sum()

0

We have 5268 duplicates in our data set so we dropped them.

In [51]:
#Changing wrong datatype
df['CustomerID']= df['CustomerID'].astype(str)
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 536641 entries, 0 to 541908
Data columns (total 9 columns):
 #   Column       Non-Null Count   Dtype         
---  ------       --------------   -----         
 0   InvoiceNo    536641 non-null  object        
 1   StockCode    536641 non-null  object        
 2   Description  535187 non-null  object        
 3   Quantity     536641 non-null  int64         
 4   UnitPrice    536641 non-null  float64       
 5   Total        536641 non-null  float64       
 6   InvoiceDate  536641 non-null  datetime64[ns]
 7   CustomerID   536641 non-null  object        
 8   Country      536641 non-null  object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(5)
memory usage: 40.9+ MB


The CustomerID column was float, and we won't make any calculations on it, so we changed it to string.

In [52]:
#Dropping quanities less than 1
df.drop(df[df['Quantity'] <= 1].index, inplace = True)

There are some negative values in the "Quantity" column, so we decided to drop them.

In [53]:
#Dropping unit prices less than 0
df.drop(df[df['UnitPrice'] <= 0].index, inplace = True)

There are some negative values in the "UnitPrice" column, so we decided to drop them.

In [54]:
#Dropping Totals less than 0
df.drop(df[df['Total'] <= 0].index, inplace = True)

There are some negative values in the "Total" column, so we decided to drop them.

In [57]:
#Showing some descriptive statistics about the dataset after dropping
df.describe()

Quantity      UnitPrice          Total
count  380783.000000  380783.000000  380783.000000
mean       14.255692       2.791756      25.332780
std       183.350748       3.419901     316.036046
min         2.000000       0.040000       0.120000
25%         3.000000       1.250000       6.720000
50%         6.000000       1.790000      14.850000
75%        12.000000       3.750000      20.400000
max     80995.000000     649.500000  168469.600000

In [58]:
#Save the dataset after the cleaning phase
df.to_excel('Online Retail.xlsx')

<a id='Questions for Analysis'></a>
## Questions for Analysis
1- What is the highest and the lowest countries' sales performance?    
2- What is the country's revenue performance?                                                                                         
3- Is there any sales trends over the months of the year in different countries?                                                                
4- How does the product perform in different countries?                                                                                       
5- Does the sold quantity of a product affects its revenue?

<a id='Data Visualization'></a>
## Data Visualization
#### Power BI Report Link:
https://www.novypro.com/project/data-visualization-project-

<a id='conclusions'></a>
## Conclusion

- From Question(1): The top 5 countries which are performing well in sales are Netherlands, Eire, Germany, France and Australia, so they can expand their business in these countries.
- From Question(2): The median of the country's revenue is 17.7 Million dollars, so as they said their business is performing well, and the top 5 countries in revenue are the same as the countries performing well in sales.
- From Question(3): The trends time is different between the countries, so depending on each country the stakeholders can take decisions to prepare their stores in trends time.
- From Question(4): There are some products which bring high revenue such as the products with stock codes "Post" and "23084", on another hand There are some products bring very low revenue such as the products with stock codes "21306" and "21034", so they should enhance their products with lower revenue, in addition to they should care about their marketing, and keep the quality of the products which bring high revenue.
- From Question(5): There is a positive correlation between the sold quantity of a product and its revenue, so they should care about selling more quantity of the products.


### Limitations
     
After our analysis, We didn't observe any problem that totally obstructs our analysis. Even the NULL values that we found on the "Description" and "CustomerID" columns didn't affect our analysis. But we faced a single challenge which is:            
There are some negative values in "Quantity", "UnitPrice" and "Total" columns, so we decided to drop them and this affect the perfectionism of our analysis.                                                                                               
                                                                                                                              